cmd/xdpcap/flags.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// nopWriter implements io.Writer and discards everything written to it
+type nopWriter struct {
+}
+
+func (nopWriter) Write(p []byte) (int, error) {
+	return len(p), nil
+}
+
+type flags struct {
+	*flag.FlagSet
+	name string
+
+	perCPUBuffer uint
+	watermark    uint
+	quiet        bool
+	flush        bool
+}
+
+func newFlags(name string) *flags {
+	fs := flag.NewFlagSet(name, flag.ContinueOnError)
+	// flag insists on always printing things out in the case of errors, disable that
+	fs.SetOutput(nopWriter{})
+
+	flags := flags{
+		FlagSet: fs,
+		name:    name,
+	}
+
+	flags.UintVar(&flags.perCPUBuffer, "buffer", 8192, "Per CPU perf buffer size to create (`bytes`)")
+	flags.UintVar(&flags.watermark, "watermark", 4096, "Perf watermark (`bytes`)")
+	flags.BoolVar(&flags.quiet, "q", false, "Don't print statistics")
+	flags.BoolVar(&flags.flush, "flush", false, "Flush pcap data written to <output> for every packet received")
+
+	return &flags
+}
+
+func (flags *flags) Usage() string {
+	usage := strings.Builder{}
+
+	usage.WriteString(fmt.Sprintf(
+		`%s [options] <debug map> <output> [<tcpdump filter expr>]
+
+Capture packets from XDP programs matching a tcpdump filter expression.
+
+<output> may be "-" to write to stdout. Implies -q and -flush.
+
+`, flags.name))
+
+	// temporarily re-enable printing to get the default usage
+	flags.FlagSet.SetOutput(&usage)
+	flags.PrintDefaults()
+	flags.FlagSet.SetOutput(nopWriter{})
+
+	return usage.String()
+}
+
+func (flags *flags) Parse(args []string) (*captureOpts, error) {
```

Comment:
This should return `config` of some sort

Reply:
`flags.Parse` has been with `newFlags()` into `parseFlags()`, which just returns a `flags`.